app/src/main/java/com/example/sudoku/ScoreboardEntryModel.kt

```diff
@@ -1,4 +1,15 @@
 package com.example.sudoku
 
-data class ScoreboardEntryModel(val rank: String, val playerName: String, val score: String) {
+import com.google.firebase.database.IgnoreExtraProperties
+
+@IgnoreExtraProperties
+data class ScoreboardEntryModel(val rank: String = "", val playerName: String = "", val score: String = "") {
```

Comment:
I would add an ID here for the device, so in the way we configured it, the user can register himself in the scoreboard with whatever nickname he wants. If we add the ID on top of that, we can highlight hI would add an ID here for the device, so in the way we configured it, the user can register himself in the scoreboard with whatever nickname he wants. If we add the ID on top of that, we can highlight his score entries on the scoreboard, no mattering what nickname he used. A somewhat good ID I think could be a hash on the MAC address of the device. It's not perfect, but good enough I think.is score entries in the scoreboard, not mattering what the nickname he used. A somewhat good ID I think could be a hash on the MAC address of the device. It's not perfect, but good enough I think.

Reply:
The main idea sounds nice and solid, but it makes the implementation more complex. I would delay it for another PR and see if we have enough time to add it to the project.
Until then, I think we can ask the user for a nickname every time when they submit new high scores (like in old arcade games).